### PR TITLE
Loadout drop i18n

### PR DESCRIPTION
**Pull request by ericnelson0**
image

Only use available classes in the class options and translate the class names as well. 

Issue #1178

### CODE REVIEW

File: app/scripts/dimApp.i18n.js [hunk 1]
```diff
@@ -563,6 +564,7 @@
             Missing1: "Es fehlen legendäre oder exotische Gegenstände, um ein vollständiges Set zu generieren!",
             Missing2: "Es fehlen seltene, legendäre oder exotische Gegenstände, um ein vollständiges Set zu generieren!" },
           Loadouts: {
+            Any: "ANNNY",
```
Comment (bhollis) [inline app/scripts/dimApp.i18n.js]: Not sure this is proper German grammar.

File: app/scripts/loadout/dimLoadout.directive.js [hunk 2]
```diff
@@ -56,19 +56,13 @@
...
-        value: 1
-      }, {
-        label: 'Hunter',
-        value: 2
-      }];
+      vm.classTypeValues = [{ label: $translate.instant('Loadouts.Any'), value: -1 }];
+
+      scope.$on('dim-stores-updated', function(evt, data) {
+        _.each(_.reject(data.stores, function(s) { return s.isVault; }), function(store) {
+          vm.classTypeValues.push({ label: store.className, value: store.classType });
```
Comment (bhollis) [inline app/scripts/loadout/dimLoadout.directive.js]: This seems like it'll keep getting longer and longer as `dim-stores-updated` keeps getting fired. Probably better to build a new list and assign it each time.

File: app/scripts/dimApp.i18n.js (hunk 1 above)
Comment (ericnelson0) [inline app/scripts/dimApp.i18n.js]: Thanks. Missed that.

File: app/scripts/loadout/dimLoadout.directive.js (hunk 2 above)
Comment (ericnelson0) [inline app/scripts/loadout/dimLoadout.directive.js]: Nice catch!

**Comment by ericnelson0** [comment]
@delphiactual We do need the "Any" text translated for all languages in this PR.

File: app/scripts/loadout/dimLoadout.directive.js [hunk 3]
```diff
@@ -56,19 +56,12 @@
...
-        label: 'Titan',
-        value: 1
-      }, {
-        label: 'Hunter',
-        value: 2
-      }];
+      scope.$on('dim-stores-updated', function(evt, data) {
+        vm.classTypeValues = [{ label: $translate.instant('Loadouts.Any'), value: -1 }];
+        _.each(_.reject(data.stores, function(s) { return s.isVault; }), function(store) {
+          vm.classTypeValues.push({ label: store.className, value: store.classType });
```
Comment (bhollis) [inline app/scripts/loadout/dimLoadout.directive.js]: What if somebody has two warlocks? We should probably make a uniquified list and use that.

Comment (ericnelson0) [inline app/scripts/loadout/dimLoadout.directive.js]: Ah. Good point.

**Comment by delphiactual** [comment]
@ericnelson0 no, we do not need all the translations before the fix is pushed; we could actually probably substitute `Vendor.All` in

**Comment by ericnelson0** [comment]
I did see that translation, but it might have different meanings in different languages. 

I'll leave that up to everyone.

**Comment by simonefranza** [comment]
Any in Italian is "Qualsiasi"

**Comment by ericnelson0** [comment]
This is ready to go

**Pull request merged**